cylc/flow/platforms.py

```diff
@@ -184,6 +184,16 @@ def platform_from_name(
                 bad_hosts=bad_hosts
             )
 
+    for platform_name_re in list(platforms):
+        if (
+            # If the platform_name_re contains special regex chars
+            re.escape(platform_name_re) != platform_name_re
+            and re.match(platform_name_re, 'localhost')
+        ):
+            raise PlatformLookupError(
+                f'"localhost" settings will be defined by '
+                f'global.cylc[platforms][{platform_name_re}]'
+            )
```

Comment:
This message is a little cryptic, how about something like:

```
The "localhost" platform cannot be defined using a regular expression.
See the documentation for "global.cylc[platforms][localhost]" for more information.
```

With an accompanying note in `global.cylc[platforms][localhost]` containing an example like this:

```
[platforms]
    [[localhost|cylc-server-\d]]  # error
    [[localhost, cylc-server-\d]]  # ok
```